services/src/layers/postgres_layer_db.rs

```diff
@@ -617,6 +619,216 @@ where
         })
     }
 
+    async fn get_search_capabilities(&self) -> Result<SearchCapabilities> {
+        Ok(SearchCapabilities {
+            search_types: SearchTypes {
+                fulltext: true,
+                prefix: true,
+            },
+            autocomplete: true,
+            filters: None,
+        })
+    }
+
+    #[allow(clippy::too_many_lines)]
+    async fn search(
+        &self,
+        collection_id: &LayerCollectionId,
+        search: SearchParameters,
+    ) -> Result<LayerCollection> {
+        let collection = Uuid::from_str(&collection_id.0).map_err(|_| {
+            crate::error::Error::IdStringMustBeUuid {
+                found: collection_id.0.clone(),
+            }
+        })?;
+
+        let conn = self.conn_pool.get().await?;
+
+        let stmt = conn
+            .prepare(
+                "
+        SELECT DISTINCT name, description, properties
```

Comment:
why is the distinct needed here? id is the primary key right?

Reply:
I mindlessly copied it from `load_layer_collection`. IMO it can be removed in both instances

Reply:
https://github.com/geo-engine/geoengine/pull/908/commits/ebe02f98f6fa21ab75fa0a1ece78b562a8614a88